libs/common/RespReadUtils.cs

```diff
@@ -4,156 +4,372 @@
 using System;
 using System.Buffers;
 using System.Buffers.Text;
-using System.Diagnostics;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 using System.Text;
+using Garnet.common.Parsing;
 
 namespace Garnet.common
 {
     /// <summary>
-    /// Utilities for reading RESP protocol
+    /// Utilities for reading RESP protocol messages.
     /// </summary>
     public static unsafe class RespReadUtils
     {
         /// <summary>
-        /// Get Header length
+        /// Tries to read the leading sign of the given ASCII-encoded number.
         /// </summary>
-        /// <param name="len"></param>
-        /// <param name="ptr"></param>
-        /// <param name="end"></param>
-        /// <returns></returns>
-        public static bool ReadHeaderLength(out int len, ref byte* ptr, byte* end)
+        /// <param name="ptr">String to try reading sign from.</param>
+        /// <param name="negative">Whether the sign is '-'.</param>
+        /// <returns>True if either '+' or '-' was found, false otherwise.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryReadSign(byte* ptr, out bool negative)
         {
-            len = -1;
-            if (ptr + 3 >= end)
-                return false;
+            negative = (*ptr == '-');
+            return negative || (*ptr == '+');
+        }
 
-            Debug.Assert(*ptr == '$');
-            ptr++;
-            bool neg = *ptr == '-';
-            int ksize = *ptr++ - '0';
-            while (*ptr != '\r')
+        /// <summary>
+        /// Tries to read an unsigned 64-bit integer from a given ASCII-encoded input stream.
+        /// </summary>
+        /// <param name="ptr">Pointer to the beginning of the ASCII encoded input string.</param>
+        /// <param name="end">The end of the string to parse.</param>
+        /// <param name="value">If parsing was successful, contains the parsed ulong value.</param>
+        /// <param name="bytesRead">If parsing was successful, contains the number of bytes that were parsed.</param>
+        /// <returns>
+        /// True if a ulong was successfully parsed, false if the input string did not start with
+        /// a valid integer or the end of the string was reached before finishing parsing.
+        /// </returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryReadUlong(ref byte* ptr, byte* end, out ulong value, out ulong bytesRead)
```

Comment:
Big fan of the``TryRead`` prefix pattern 👍 

Can we do it for the other overloads too that return ``bool`` and have a ``byref`` result, that is ``out T value``?

Reply:
I fully agree! I think we should have a separate cleanup PR to unify the naming of all parsing related functions. As for BCL naming, I personally find BCL naming clearer for integers, but we should be consistent across the whole codebase.